test/mpi/preconditioner/schwarz.cpp

```diff
@@ -264,6 +276,22 @@ TYPED_TEST(SchwarzPreconditioner, CanApplyPreconditioner)
 }
 
 
+TYPED_TEST(SchwarzPreconditioner, CanApplyMultilevelPreconditioner)
```

Comment:
The test is not enough because it does not verify the result based on something we tested before.
You can do the workflow again like you did in core. also need two coarse weight testcase: one is in the [0, 1] and the other is default.

I was thinking whether extracting the same thing from core to here is considered valid or not.
It seems to be duplication but the routines here can be considered verified by the other tests.

Reply:
Which core tests do you mean ? I dont think there are any tests in core which do a proper apply.

Reply:
Should be updated now. I just use Multigrid as a preconditioner to emulate the coarse apply and a local precond separately and then add the solutions together.